.github/workflows/table_of_contents.yml

```
@@ -0,0 +1,76 @@
+# This GitHub Action is used for triggering updates of
+# the toc.json files present in any directory that
+# needs an automatically generated table of contents.
+
+name: Generate Table of Contents files
+
+env:
+  # Force the stdout and stderr streams to be unbuffered
+  PYTHONUNBUFFERED: 1
+
+on:
+  pull_request:
+    types:
+      - synchronize
+      - reopened
+      - opened
+permissions:
+  contents: write
+jobs:
```

let's try this to rule out permission issues

```suggestion
      - opened

permissions: write-all

jobs:
```